Configurations/10-main.conf

```diff
@@ -1157,6 +1160,7 @@ my %targets = (
         bn_ops           => "SIXTY_FOUR_BIT_LONG RC4_CHAR",
         perlasm_scheme   => "aix64",
         shared_ldflag    => add_before("-shared -static-libgcc"),
+        shared_extension => "64.so.\$(SHLIB_VERSION_NUMBER)",
```

Comment:
Hmm, re `64`, have a look in the top `build.info`

Reply:
Could you be more specific? Do you suggest to handle it there instead of here? For reference, goal is to have libcrypto.a with libcrypto.so.1.1 and libcrypto64.so.1.1 in it.

Reply:
> Do you suggest to handle it there instead of here?

I.e. in code instead of parameter?

Reply:
Yes. As you can see, all others that encode bitness in the library file name do it there. So doing so for AIX as well would be consistent...

Reply:
I'd like to argue against this suggestion. Why. First of all I'd like to say that I **feel** that you are being somewhat inconsistent. What was your first comment? "I thoroughly hate this". But what was it? You hated use of code to solve specific variable assignment problem. But doesn't top of build.info do pretty much same thing? I.e. uses code to solve specific variable assignment problem? It does. Then why hate one, but not another? Because it's in one place? Trouble here is that it's in one place in *your* mind. In my mind it's just an odd(*) place among unknown number of as odd places where things can happen. One can also put it as following, it's code-oriented *implicit* variable assignment (using unspecified amount of inputs) vs. table-driven *explicit* variable assignment. I'd argue that whenever possible, table-driven assignment should be preferred. Because it's more transparent to everybody involved. So that having said this, I'd actually argue rather in favour of *omitting* referred code from beginning of build.info and making arrangements that would allow to switch to table-driven assignment. To be more specific I'd find `shared_extension => "-$(DLL_VERSION_NUMBER)-x64.dll"` more appropriate than it being synthesized purely in scattered code as it is now. [Yes, it means that there would still be code somewhere that would assign DLL_VERSION_NUMBER, but it would be unconditional assignment.] Why more appropriate? For example because it user wishes to arrange a variant dll, it would be just a matter of overriding a variable in .conf file. As opposite to navigating among perl snippets in custom-language file such as build.info.

(*) Why odd? Because I can't let be wondering it it's right spot. For example, why would vms-specific variable assignment have to be done in build.info and not in descrip.mms.tmpl? I mean wouldn't it be nice if there was *one* place where one solves vms-specific corner problems?

Reply:
> For example because it user wishes to arrange a variant dll, it would be just a matter of overriding a variable in .conf file.

One can naturally argue that in Windows case it could be achieved by manipulating multilib attribute. But thing is that it's just too far from obvious to unprepared user. Because "multilib" is not commonly recognized concept, especially to Windows user, so it wouldn't give right associations. While very same user would have to problem whatsoever recognizing something that ends with .dll and associate it with corresponding file name extension, even though attribute is named "shared_extension." I mean "shared_extension" name doesn't *directly* associate with "dynamically-loaded library filename extension," one as well initially imagine that it's some kind of extension that is shared among something.

Reply:
The reason to have the shared library *name* in `build.info` is to make that information available through config data, and usable by anything else where we might need it, while the shared library *extension* is handled by the system loading the shared library (be it `dlopen`, `lib$find_image_symbol` or what not).  If you have a grep of `sharednames`, you will see that it's currently used in `test/recipes/90-test_shlibload.t`, `util/mkdef.pl`, `util/local_shlib.com.in` and `util/unlocal_shlib.com.in`.

If you move all those parts that we use to make up the shared library file *name* to the build files, you will also force us to reproduce the exact same mechanisms in other places.  This leads to a maintenance burden because of duplication of effort.

*Please do not do this*

Reply:
However, AIX treats the 64 as extension and not as part of the name. You still link with -lcrypto and the result is a binding to libcrypto.a(libcrypto64.so.1.1).

Reply:
Ah ! Yeahok. Sorry for the noise then, haven't quite wrapped my brain around the AIX way, it seems

Reply:
> The reason to have the shared library name in build.info is to make that information available through config data,

I don't quite follow. The shared library name is synthesized in build.info based in config data. It's synthesized based on explicitly programmed VC/mingw/cygwin/vms rules. I see no reason for why one can't omit these rules(*) and replace them with single table-driven assignment... Maybe I'm missing something... However! Do note that I wrote "I **would** argue rather in favour of omitting referred code," i.e. I'm not suggesting to actually do it [at least not at this point], only speak in favour of preferring and facilitating(**) table-driven approach [as more transparent] whenever possible. And suggested approach does work (note that it's not about library name, but in this case **intermediate** shared library extension), and I see no reason to complicate build.info with AIX-specific code.

(*) Or at least most of them. It seems that it would be impossible to omit Cygwin-specific assignment, because it's about prefix, but as far as suffices go, they can come from $config...

(**) Configure modification in this request are exactly that, they facilitate table-driven assignment.

Reply:
I'm not sure what table driven assignment would look like, care to give a quick example?  You *are* talking about another attribute in the config targets, aren't you?  I'm not against that idea, but would like to see what you'd like to propose.

Reply:
??? The disputed line is the example. It reads
```
shared_extension => "64.so.\$(SHLIB_VERSION_NUMBER)",
```
and is repeated in aix64-\* targets. It overrides .so.\$(SHLIB_VERSION_NUMBER) set in aix-common. You reaction was "this should go to top of build.info". My interpretation was that injection of "64" prior .so is to be implemented as code snippet, naturally conditional on target name starting with aix64. It doesn't have to, direct assignment above works in this situation. And assertion is that the fact that it works should make it preferred.

[In the course of discussion I also suggested that assigning "-$(DLL_VERSION_NUMBER)-x64.dll" to corresponding argument, i.e. shared_extension, in VC and mingw targets, **would** be more transparent to everybody involved. More transparent than it is now, when string is composed programmatically in build.info.]

Reply:
@makr's last remark nullified my concern regarding `64`...  My *last* remark was in response to you talking about having things table driven instead of the current platform-specific SHARED_NAME stuff in the top build.info.  That should probably be discussed elsewhere, however.

Reply:
> @makr's last remark nullified my concern regarding 64 ...

So...

Reply:
So you can ignore my initial concern here

Reply:
I rather meant "so... approve then?" [Or at least dismiss change request.]